### PR TITLE
+htc add missing javadsl.Http.outgoingConnectionTls overload + remove extra params

### PR DESCRIPTION
**Pull request by jrudolph**

### CODE REVIEW

File: akka-http-core/src/main/scala/akka/http/javadsl/Http.scala [hunk 1]
```diff
@@ -330,7 +333,6 @@ class Http(system: ExtendedActorSystem) extends akka.actor.Extension {
    * for encryption on the connection.
    */
   def newHostConnectionPoolTls[T](host: String, port: Int,
-                                  options: JIterable[Inet.SocketOption],
```
Comment (ktoso) [inline akka-http-core/src/main/scala/akka/http/javadsl/Http.scala]: is there an overload with these options already?

Comment (jrudolph) [inline akka-http-core/src/main/scala/akka/http/javadsl/Http.scala]: No, at some point I moved the options into the settings but it seems I forgot to remove all the parameters I had added before.

Comment (ktoso) [inline akka-http-core/src/main/scala/akka/http/javadsl/Http.scala]: I see. Makes sense

**Comment by ktoso** [comment]
Thanks!

**Pull request merged**